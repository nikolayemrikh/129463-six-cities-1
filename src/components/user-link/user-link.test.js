import React from 'react';
import renderer from 'react-test-renderer';

import UserLink from './user-link.jsx';

const mock = {
  email: `mock@mock.com`,
  avatarSrc: `src`,
};

describe(`UserLink compontent`, () => {
  it(`should render correctly with email and avatar if user prop presented`, () => {
    const tree = renderer
      .create(<UserLink user={mock}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should render correctly with sign in button if user prop not presented`, () => {
    const tree = renderer
      .create(<UserLink/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
