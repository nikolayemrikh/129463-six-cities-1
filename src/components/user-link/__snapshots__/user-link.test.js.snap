// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserLink compontent should render correctly with email and avatar if user prop presented 1`] = `
<a
  className="header__nav-link header__nav-link--profile"
  href="#"
>
  <div
    className="header__avatar-wrapper user__avatar-wrapper"
  >
    <img
      src="src"
    />
  </div>
  <span
    className="header__user-name user__name"
  >
    mock@mock.com
  </span>
</a>
`;

exports[`UserLink compontent should render correctly with sign in button if user prop not presented 1`] = `
<a
  className="header__nav-link header__nav-link--profile"
  href="#"
>
  <span
    className="header__login"
  >
    Sign in
  </span>
</a>
`;
